### PR TITLE
AddGroup Command with the GroupID as 0 now responds as INVALID_COMMAND.

### LINKED ISSUE #18508: [SQA] TC-GR-2.2 Chiptool allows Group ID ' 0 ' in Group Cluster
#### Problem
During the execution of AddGroup Command with the GroupID as 0 we should get response as INVALID_COMMAND as the Group ID should be in the inclusive range of 1 to 65535

 * expected behavior : Response should be INVALID_COMMAND
 * actual behavior : Getting Response as status: 135 (Status is enum8 type)
 * Command used : `./chip-tool any command-by-id 0x0004 0x00 '{ "0": 0, "1": "my-group-name"}' 1 0`
 * system configuration : 
- Chip-tool - RPI-4, 8GB RAM
- DUT - RPI - RPI-4, 8GB RAM 
- Network - Ble-wifi

 * Log : [Groups_Issue_TH.txt](https://github.com/project-chip/connectedhomeip/files/8709558/Groups_Issue_TH.txt)

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
#### Problem
* Fixes [#18508](https://github.com/project-chip/connectedhomeip/issues/18508): TC-GR-2.2 Chiptool allows Group ID 0 in Group Cluster

#### Change overview
* Added GroupId validation in AddGroup command callback.
* Added new test case in the Groups cluster YAML.

#### Testing
* Groups Tests executed successfully.

### CODE REVIEW

**Review by tcarmelveilleux — APPROVED** [review]

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/clusters/groups-server/groups-server.cpp [hunk 1]
```diff
@@ -79,6 +79,7 @@ static bool GroupExists(FabricIndex fabricIndex, EndpointId endpointId, GroupId
 static EmberAfStatus GroupAdd(FabricIndex fabricIndex, EndpointId endpointId, GroupId groupId, const CharSpan & groupName)
 {
     VerifyOrReturnError(IsFabricGroupId(groupId), EMBER_ZCL_STATUS_INVALID_VALUE);
+    VerifyOrReturnError(IsValidGroupId(groupId), EMBER_ZCL_STATUS_INVALID_COMMAND);
```
[the remark this answers is not in the document]
Comment (rcasallas-silabs) [reply src/app/clusters/groups-server/groups-server.cpp:81]: IsValidGroupId condition moved first.

File: src/app/clusters/groups-server/groups-server.cpp [hunk 2]
```diff
@@ -78,6 +78,7 @@ static bool GroupExists(FabricIndex fabricIndex, EndpointId endpointId, GroupId
 
 static EmberAfStatus GroupAdd(FabricIndex fabricIndex, EndpointId endpointId, GroupId groupId, const CharSpan & groupName)
 {
+    VerifyOrReturnError(IsValidGroupId(groupId), EMBER_ZCL_STATUS_INVALID_COMMAND);
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/clusters/groups-server/groups-server.cpp:81]: Well actually no it is now worse.... it's `CONSTRAINT_ERROR` as per : https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/5314/files

[the remark this answers is not in the document]
Comment (rcasallas-silabs) [reply src/app/clusters/groups-server/groups-server.cpp:81]: CONSTRAINT_ERROR is an alias for the now deprecated INVALID_VALUE, so these changes are no longer needed. However, I will replace the old name in the code.

[the remark this answers is not in the document]
Comment (rcasallas-silabs) [reply src/app/clusters/groups-server/groups-server.cpp]: Spec change: The correct code is now CONSTRAINT_ERROR.

File: src/app/clusters/groups-server/groups-server.cpp [hunk 3]
```diff
@@ -78,7 +78,7 @@ static bool GroupExists(FabricIndex fabricIndex, EndpointId endpointId, GroupId
 
 static EmberAfStatus GroupAdd(FabricIndex fabricIndex, EndpointId endpointId, GroupId groupId, const CharSpan & groupName)
 {
-    VerifyOrReturnError(IsFabricGroupId(groupId), EMBER_ZCL_STATUS_INVALID_VALUE);
+    VerifyOrReturnError(IsFabricGroupId(groupId), EMBER_ZCL_STATUS_INVALID_COMMAND);
```
[the remark this answers is not in the document]
Comment (rcasallas-silabs) [reply src/app/clusters/groups-server/groups-server.cpp]: It is now.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by bzbarsky-apple — COMMENTED** [review]
All the endpoint_config.h changes in this PR should be removed, right?  @rcasallas-silabs

**Pull request merged**